crates/swc_ecma_parser/src/parser/ident.rs

```diff
@@ -139,8 +138,8 @@ impl<I: Tokens> Parser<I> {
 
                 // It is a Syntax Error if the goal symbol of the syntactic grammar is Module
                 // and the StringValue of IdentifierName is "await".
-                Word::Keyword(Keyword::Await) if p.ctx().module => {
-                    syntax_error!(p, p.input.prev_span(), SyntaxError::ExpectedIdent)
+                Word::Keyword(Keyword::Await) if p.ctx().module | p.ctx().in_async => {
```

Comment:
https://github.com/swc-project/swc/blob/03a2fd06ce548214cb75f51d0ea47795a24c6cce/crates/swc_ecma_parser/src/parser/ident.rs#L135-L137

@kdy1 Do you have any idea about the error is ignored in typescript mode?

Reply:
I think there was a test case in `tsc` test list. I don't remember the name, though.
So you may need to remove this check and run tsc test of the parser

Reply:
I removed these lines and got more errors.

https://github.com/swc-project/swc/blob/e06d717c36572f8e45a07a2ec5cd32171fecdf32/crates/swc/tests/tsc.rs#L187-L192

Since we always test tsc files in module mode, we may get more errors than expected.

Reply:
I see, I think it's fine for now